From: Arnd Bergmann <arnd@arndb.de>
Date: Wed, 22 Aug 2018 17:25:44 +0200
Subject: net_sched: fix unused variable warning in stmmac
Patch-mainline: v4.19-rc2
Git-commit: 191672ca07a7c10c3b84d01019a33d59b4317997
References: bsc#1109837

The new tcf_exts_for_each_action() macro doesn't reference its
arguments when CONFIG_NET_CLS_ACT is disabled, which leads to
a harmless warning in at least one driver:

drivers/net/ethernet/stmicro/stmmac/stmmac_tc.c: In function 'tc_fill_actions':
drivers/net/ethernet/stmicro/stmmac/stmmac_tc.c:64:6: error: unused variable 'i' [-Werror=unused-variable]

Adding a cast to void lets us avoid this kind of warning.
To be on the safe side, do it for all three arguments, not
just the one that caused the warning.

Fixes: 244cd96adb5f ("net_sched: remove list_head from tc_action")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/net/pkt_cls.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/net/pkt_cls.h
+++ b/include/net/pkt_cls.h
@@ -302,7 +302,7 @@ static inline void tcf_exts_put_net(stru
 	for (i = 0; i < TCA_ACT_MAX_PRIO && ((a) = (exts)->actions[i]); i++)
 #else
 #define tcf_exts_for_each_action(i, a, exts) \
-	for (; 0; )
+	for (; 0; (void)(i), (void)(a), (void)(exts))
 #endif
 
 static inline void
